### DISCUSSION TITLE
Default track per assembly

### DISCUSSION THREAD

**Discussion by FabianDK** [GMOD/jbrowse-components#3135] [Q&A]
Dear devs,

I have multiple assemblies added to JBrowse2, each of which has a gff3 track added. What I want is that if an assembly is opened, the gff3 tracks are already pre-loaded.

I found out that I can set a default session by downloading (File/Export Session option) and setting the session.json (set-default-session). However, this also sets a default assembly which I would like to avoid.

I also tried this command but get an error (example for two assemblies): 
```
# add assemblies and tracks
jbrowse add-assembly /jbrowse/genomes/GCF_003668045.3_CriGri-PICRH-1.0_genomic.fna --out /jbrowse/jbrowse2 --load copy --name PICRH_v1 --displayName "CHO reference sequence (PICRH v1.0)"
jbrowse add-track /jbrowse/genomes/GCF_003668045.3_CriGri-PICRH-1.0_genomic.sorted.gff3.gz --out /jbrowse/jbrowse2 --assemblyNames PICRH_v1 --load copy -n "Genes & transcripts (gff)" --category="Annotation" 
jbrowse text-index --out /jbrowse/jbrowse2 -a PICRH_v1

jbrowse add-assembly /jbrowse/genomes/GCF_000001635.27_GRCm39_genomic.fna --out /jbrowse/jbrowse2 --load copy --name GRCm39 --displayName "Mouse reference sequence (GRCm39)"
jbrowse add-track /jbrowse/genomes/GCF_000001635.27_GRCm39_genomic.sorted.gff3.gz --out /jbrowse/jbrowse2 --assemblyNames GRCm39 --load copy -n "Genes & transcripts (gff)" --category="Annotation"
jbrowse text-index --out /jbrowse/jbrowse2 -a GRCm39

# set default session (not working)
jbrowse set-default-session --target /jbrowse/jbrowse2/config.json --view LinearGenomeView --tracks GCF_003668045.3_CriGri-PICRH-1.0_genomic.sorted.gff3,GCF_000001635.27_GRCm39_genomic.sorted.gff3
```

When I then start JBrowse2, and open an assembly:
```
Fatal Error:
TypeError: Cannot read properties of undefined (reading 'height')
```

Could you please help me to resolve this?

**Comment by garrettjstevens** [comment]
I don't think what you describe is currently possible, but I want to get a couple more details to be sure.

> What I want is that if an assembly is opened, the gff3 tracks are already pre-loaded.

When you say "opened", do you mean in a Linear Genome View, or some other view, or something else? Also, by "pre-loaded", do you mean the tracks are open and visible, or that they are available in the track selector?

**Comment by FabianDK** [comment]
> When you say "opened", do you mean in a Linear Genome View, or some other view, or something else? 

I mean after I select Linear Genome View, I come to the window where I can choose the added assemblies and open them by clicking the OPEN button.

> Also, by "pre-loaded", do you mean the tracks are open and visible, or that they are available in the track selector?

The tracks are already available in the track selector, so what I would like is that the tracks are open and visible. In our case, basically every user would be interested in the gff track. Having the gff tracks open and visible after a user has opened the assembly of interest would avoid some potential confusion and requires less clicking around.
